exonum-java-binding/qa-service/src/main/java/com/exonum/binding/qaservice/ApiController.java

```diff
@@ -333,15 +335,17 @@ private void failureHandler(RoutingContext rc) {
 
     Throwable requestFailure = rc.failure();
     if (requestFailure != null) {
-      Optional<String> badRequest = badRequestDescription(requestFailure);
-      if (badRequest.isPresent()) {
-        respondBadRequest(rc, badRequest.get());
+      HttpServerResponse response = rc.response();
+      if (isBadRequest(requestFailure)) {
+        logger.info("Request error:", requestFailure);
+        response.setStatusCode(HTTP_BAD_REQUEST);
       } else {
-        logger.error("Request error:", requestFailure);
-        rc.response()
-            .setStatusCode(HTTP_INTERNAL_ERROR)
-            .end();
+        logger.error("Internal error", requestFailure);
+        response.setStatusCode(HTTP_INTERNAL_ERROR);
       }
+      String description = Strings.nullToEmpty(requestFailure.getMessage());
+      response.putHeader("Content-Type", "text/plain")
```

Comment:
why do we return plain test for errors? It'll complicate clients because they need to handle json for ok responses and test for errors.

Reply:
@dip56 Does it complicate your client? If it doesn't, I would not bother changing that.

Reply:
It doesn't matter to me, but I think it is a good idea to return all errors in a single JSON format.

Reply:
https://jira.bf.local/browse/ECR-3123

Reply:
It is, but as a separate task.